### PR TITLE
Update messaging from unit conversion plugin

### PR DESCRIPTION
**Pull request by cshanahan1**
This PR improves the messaging from the unit conversion plugin to broadcast changes in selected display unit. Because some plugins are listening for changes to the spectral y axis (either from toggling flux<>sb, changing flux or angle when it is toggled to SB, or changing flux when it is toggled to flux) this is its own message now.

Every time a new selection of 'flux' is made, there are three GlobalDisplayUnitChanged messages broadcasted : 'flux', 'sb', and 'spectral_y'.

While this is not included in this PR, these changes will make it possible for the correct messaging to be broadcasted when a new angle selection is made (always sending a 'sb' message, and if the spectral y axis is in SB also sending a 'spectral_y' message).

Every time there is a toggle between flux<> sb, the GlobalDisplayUnitChanged axis='spectral_y' message is broadcasted.

A follow up in https://github.com/spacetelescope/jdaviz/pull/3144 will fix how plugins respond to this updated messaging - for now, with the additional broadcasts, there will be some redundant unit conversions in plugins. I fixed some instances of this (in coords_info, spectral extraction) but the follow up for other plugins to intercept the correct message will be in that PR.

### CODE REVIEW

File: CHANGES.rst
Comment (kecnry) [inline CHANGES.rst:137]: maybe this PR number can just be appended to the existing changelog entry for unit conversion?

File: jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py
Comment (kecnry) [inline jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py:330]: this then needs to also make a call to `_update_results_units`/`_update_units_on_function_selection` (in which case maybe it shouldn't be renamed), right?
```suggestion
else:
    return
self._update_units_on_function_selection(msg)
```

File: jdaviz/configs/default/plugins/model_fitting/model_fitting.py
Comment (kecnry) [inline jdaviz/configs/default/plugins/model_fitting/model_fitting.py:531]: see https://github.com/spacetelescope/jdaviz/pull/3144#discussion_r1718787001

File: jdaviz/configs/imviz/plugins/coords_info/coords_info.py
Comment (kecnry) [inline jdaviz/configs/imviz/plugins/coords_info/coords_info.py:126]: I think this can be removed entirely for now and leave any other special-handling up to @gibsongreen's effort to enable in the other configs.  _Hopefully_ we can write the logic in a way that we don't need per-config handling 🤞

File: jdaviz/configs/imviz/plugins/coords_info/coords_info.py
Comment (kecnry) [inline jdaviz/configs/imviz/plugins/coords_info/coords_info.py:135]: should we just leave this out for now if we know we'll immediately remove it?  If so, this whole diff for this method just becomes "flux" -> "sb"

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (kecnry) [inline jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:248]: see https://github.com/spacetelescope/jdaviz/pull/3144#discussion_r1718789266

File: jdaviz/configs/imviz/plugins/coords_info/coords_info.py
Comment (gibsongreen) [reply jdaviz/configs/imviz/plugins/coords_info/coords_info.py:126]: This is one of the spots where I added the older logic into #3139, I can test this PR with and without to see the implications, I think the only config that could need specific handling (not necessarily here) is Imviz: https://github.com/spacetelescope/jdaviz/blob/e29478e83b9fc5cd4568f0b5c290f67e80e7e7df/jdaviz/configs/imviz/plugins/coords_info/coords_info.py#L124

File: jdaviz/configs/imviz/plugins/coords_info/coords_info.py
Comment (cshanahan1) [reply jdaviz/configs/imviz/plugins/coords_info/coords_info.py:135]: if i remove this, it will break for cubes loaded in flux units until i add the logic for converting between square pixels and angles. if thats ok with you i can remove it

Comment (kecnry) [reply jdaviz/configs/imviz/plugins/coords_info/coords_info.py:135]: Either way is fine with me, as long as it gets simplified back to the one liner in the end instead of the double casting to a unit

File: jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py
Comment (cshanahan1) [reply jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py:330]: I don't think  _update_units_on_function_selection needs to be called here. if the flux display unit is changed, the 'flux_unit' and 'sb_unit' traitlet will be updated. the function selection determines which one of these becomes the display unit, but both are available and updated to the new unit

Comment (cshanahan1) [reply jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py:330]: i don't think this _update_units_on_function_selection needs to be called again, but correct me if i'm wrong.

upon selection of a new 'flux' unit in UC, both the flux_unit and sb_unit traitlets will be updated in the spectral extraction plugin. which of these is used for the display unit is determined by choice of 'function', but it has access to both which are synced with the current selection in unit conversion now.

File: jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py
Comment (cshanahan1) [reply jdaviz/configs/specviz/plugins/unit_conversion/unit_conversion.py:248]: done, i changed the variable name to spectral_y and set it to self.flux_or_sb

File: jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py
Comment (kecnry) [reply jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py:330]: But those traitlets aren't observed in that method and the string itself is set, not a reference. I can test in the morning to see what actually happens though, I only had time to skim the code.

File: jdaviz/configs/imviz/plugins/coords_info/coords_info.py
Comment (cshanahan1) [reply jdaviz/configs/imviz/plugins/coords_info/coords_info.py:126]: it will for sure need to be changed for imviz, i can't just intercept a sb changed message for imviz. i can remove it for now though i think?

File: jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py
Comment (kecnry) [reply jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py:330]: Tested and this does not currently work, but didn't work in main either and is technically covered by another ticket, so probably can be deferred until then so we can get the rest of this in asap.

File: jdaviz/configs/default/plugins/model_fitting/model_fitting.py
Comment (kecnry) [inline jdaviz/configs/default/plugins/model_fitting/model_fitting.py:532]: does this work?  Aren't the message axis values either `spectral` or `spectral_y` (not `spectral_x` or `spectral_y`)?

Comment (cshanahan1) [reply jdaviz/configs/default/plugins/model_fitting/model_fitting.py:532]: whoops yeah, ill change that

**Review by kecnry — APPROVED** [review]
I think this is good to build on top of now - thanks!  (Obviously will need the follow-ups for non-steradian support, fixing previews in spectral extraction, and anything remaining from #3144).

**Comment by cshanahan1** [comment]
@kecnry i just added a call to update based on function selection, does that fix the issue you were seeing?

**Comment by kecnry** [comment]
I don't think it fully fixes it, but I have that ticket so can take over once this is merged.  Since it didn't break here, I think it can be considered out of scope

**Review by gibsongreen — APPROVED** [review]
These changes look good and to me and should be really helpful for the rest of the effort, great work!

**Pull request merged**
